graph/community/louvain_directed.go

```diff
@@ -353,25 +352,25 @@ func (g *ReducedDirected) To(v graph.Node) []graph.Node {
 func (g *ReducedDirected) HasEdgeBetween(x, y graph.Node) bool {
 	xid := x.ID()
 	yid := y.ID()
-	if xid == yid {
+	if xid == yid || xid != int64(int(xid)) || yid != int64(int(yid)) {
```

Comment:
The communities are internal representations with dense IDs stored in memory; this means that an `int` is large enough. It has additional benefits performance-wise in that we are dealing with machine word-sized values on all archs instead of pushing `int64`s through 386 and arm. These are the people we are making the `int`->`int64` ID change for. In addition, the code changes required for this approach are far less wide-reaching than the straight out mechanical change.